src/calibre/utils/ipc/__init__.py

```diff
@@ -44,7 +44,7 @@ def gui_socket_address():
             if not user:
                 user = os.path.basename(os.path.expanduser('~'))
             if islinux:
-                ADDRESS = (u'\0%s-calibre-gui.socket' % ascii_filename(force_unicode(user))).encode('ascii')
+                ADDRESS = (u'\0%s-calibre-gui.socket' % ascii_filename(force_unicode(user)).decode('ascii'))
```

Comment:
Actually this is wrong as it only works on python3, and the other way only worked on python2 (because the multiprocessing module specifically checks for identity as a `str()` and now instead of being bytes on python3, it is unicode on python2)

Reply:
@kovidgoyal am I correct in thinking that ascii_filename() is intended to return `bytes()`?